src/main/java/seedu/address/model/util/DateTime.java

```diff
@@ -0,0 +1,83 @@
+package seedu.address.model.util;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+
+public class DateTime implements Comparable<DateTime> {
+    public static final String TIME_REGEX = "(([0-1]\\d)|(2[0-3])):([0-5]\\d)";
+    public static final String DATE_REGEX = "(([0-2]\\d)|(3[0-1]))-((0[1-9])|(1[0-2]))-(\\d{4})";
+    public static final String MESSAGE_CONSTRAINTS =
+            "From should be in the format of DD-MM-YYYY or DD-MM-YYYY HH:mm, and should not be blank."
+                    + "Note: Single digit month, day, and minute must start with a leading zero.";
+
+    public static final String VALIDATION_REGEX = String.format("%s(\\s(%s))?",
+            DATE_REGEX, TIME_REGEX);
+
+    public static final DateTimeFormatter DATE_TIME_FORMATTER =
+            DateTimeFormatter.ofPattern("dd-MM-yyyy[ HH:mm]");
+
+    public final LocalDateTime value;
+    public final String valueString;
```

Comment:
I was thinking whether we could just generate the valueString on the go instead of having an additional attribute

Reply:
I think there are some challenges with this cos we would then need to check if localDateTime is meant to be only a date or there is a time component also.

Reply:
@shadowezz is correct. It would be easier to generate the valueString when instantiating. Anyways the value is final so the valueString doesn't change